source/include/core_json.h

```diff
@@ -155,15 +156,17 @@ JSONStatus_t JSON_Validate( const char * buf,
  *         value[ valueLength ] = save;
  *     }
  * @endcode
+ *
+ * @note The maximum index value is ~2 billion ( 2^31 - 9 ).
  */
 /* @[declare_json_search] */
 JSONStatus_t JSON_Search( char * buf,
```

Comment:
Is there a reason why the buffer isn't of a `const char *` type considering that it is an input parameter?

Reply:
Yes, since the outputs alias the input, it makes a mess for the caller w.r.t. warnings and MISRA.